feat: Added checks workflow

alexandrtovmach opened the pull request:
Workflow to validate changes and protect spec content from direct updates. Added linter for markdown and standardized contentbase.

## Why it is?

1. We should validate changes by automated tools
2. SemVer project contain two repositories, one for spec, another one for website. This PR should solve problem with direct spec updates not in a right place. All changes about SemVer should be in **semver/semver**, not here, so this workflow will warn users with message:

![message](https://user-images.githubusercontent.com/28801003/84565588-c6b74d00-ad72-11ea-9b1f-e4de053a2dae.png)

## How it works?

In case that GitHub Actions doesn't provide access to `GITHUB_TOKEN` which required for writing comment to PR, we should trigger it manually. Its normal practice at the current moment, and to trigger checks need print one comment to PR with `/check`:

![image](https://user-images.githubusercontent.com/28801003/84574788-93e07980-adb1-11ea-9589-108e7b079ede.png)

alexandrtovmach committed:
feat: Added protect-content workflow

alexandrtovmach committed:
feat: Updated checks workflow

alexandrtovmach committed:
fix: Added temp fallback for lint script

alexandrtovmach committed:
feat: Updated workflow to trigger mannually

alexandrtovmach committed:
chore: Configured linter for md

indirect reviewed (changes requested):
This changes the rendered version of index.md, with not just indentation but also content. I don't think we want an automatic formatter to change the content of the SemVer spec!

You can see the rendered diff, and the indent changes to BNF and content changes to regex, here: https://github.com/semver/semver.org/pull/266/files?short_path=d680e8a#diff-d680e8a854a7cbad6d490c445cba2eba

alexandrtovmach committed:
refactor: Standardized content

alexandrtovmach committed:
refactor: Fixed no-consecutive-blank-lines

alexandrtovmach committed:
feat: Added mdash-style rule

alexandrtovmach commented:
@indirect Yeah, it's my mistake with `regex`, not formatter :disappointed: I did global "Find & Replace" and did in wrong way. Thanks for pay attention :+1: 

Investigated a bit more I figured out with config to avoid drastic changes. Formatting changes:
- unordered-list-marker-style  to use `"-"` (see example https://github.com/semver/semver.org/pull/266/files#diff-fe45e64f3f9908cf45fb26a75a3a3a15L277-L285)
- code-block-style to use `"fenced"`, with fixed indentation :wink:  (see example https://github.com/semver/semver.org/pull/266/files#diff-963215f81f79799fe04574157522069dL153-R216)
- removed no-consecutive-blank-lines (see example https://github.com/semver/semver.org/pull/266/files#diff-d680e8a854a7cbad6d490c445cba2ebaL235)
- mdash-style to use `"―"` (see example https://github.com/semver/semver.org/pull/266/files#diff-2f728cf67420f4f4c507ccfe1cd7320dL16-L18)
- removed consecutive spaces (see example https://github.com/semver/semver.org/pull/266/files#diff-2f728cf67420f4f4c507ccfe1cd7320dL16-L18)

alexandrtovmach commented:
@indirect could you, if this PR looks okay, approve it, merge and then turn on rule to require passing workflow check for PRs to gh-pages branch? I haven't enough permissions to do that :shrug:, so that's can be done by you or just grant me admin rights if it's not a problem

alexandrtovmach commented:
NOTE: Our default branch is `gh-pages`, so please ignore `master` on screenshots

Settings -> Branches -> Edit rule (if exist) or Add rule:
![image](https://user-images.githubusercontent.com/28801003/84700636-2f155280-af5c-11ea-80fd-27042cd2977a.png)

Set "Require status checks to pass before merging" and "Require branches to be up to date before merging" to true and select our check workflow from list (should available after merge)
![image](https://user-images.githubusercontent.com/28801003/84700678-48b69a00-af5c-11ea-8f51-a05649baab5d.png)

indirect reviewed (commented):
@alexandrtovmach looks like there's a few places in the BNF fenced code block where the rendered output is indented differently now. Can you update index.md so the rendered versions stays the same?

After that, I'm ready to merge this and turn on the setting. 👍

alexandrtovmach committed:
fix: Restored original indentation in BNFG blocks

alexandrtovmach committed:
Merge branch 'gh-pages' of github.com:semver/semver.org into feat/protect-content

alexandrtovmach commented:
@indirect Done, please take a look

indirect commented:
hmm I tried to turn on the setting, but this repo has not yet seen the "check" status 😅 maybe after someone runs it?

indirect commented:
/check

alexandrtovmach commented:
@indirect it's available after merge

indirect commented:
Done!

<img width="749" alt="image" src="https://user-images.githubusercontent.com/78/84817822-6e7a8800-afca-11ea-8a64-bc11114945af.png">

alexandrtovmach commented:
@indirect -__- something wrong, it's not report the status to PR, will check

marilouzamora408-ai commented on .github/workflows/checks.yml:
```diff
@@ -0,0 +1,56 @@
+        id: comment
+        with:
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          script: return context.payload.issue.pull_request && context.payload.comment.body === '/check'
```
Thanks for su
![Screenshot_2025-10-04-01-35-17-99](https://github.com/user-attachments/assets/8a130b03-54c3-425d-a997-9b96d2f2e653)
pport
